pkg/api/apierrors/folder.go

```diff
@@ -41,7 +41,8 @@ func ToFolderErrorResponse(err error) response.Response {
 		return response.Error(http.StatusConflict, err.Error(), nil)
 	}
 
-	if errors.Is(err, dashboards.ErrFolderVersionMismatch) {
+	if errors.Is(err, dashboards.ErrFolderVersionMismatch) ||
+		k8sErrors.IsAlreadyExists(err) {
```

Comment:
Folder API Server should return 412 status code when trying to create an already existing folder: this is the same behavior from Legacy